fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/MixinServerWorld.java

```diff
@@ -0,0 +1,39 @@
+package net.fabricmc.fabric.mixin.dimension;
+
+import java.util.function.BooleanSupplier;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionType;
+
+@Mixin(ServerWorld.class)
+public abstract class MixinServerWorld {
+	@Shadow
+	@Final
+	private MinecraftServer server;
+
+	@Inject(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/world/ServerWorld;setTimeOfDay(J)V", ordinal = 0), locals = LocalCapture.CAPTURE_FAILHARD)
+	private void setTimeInAllWorlds(BooleanSupplier shouldKeepTicking, CallbackInfo info, Profiler profiler, long time) {
+		// `time` is the time of day before it's corrected to be the morning time
+		for (World world : server.getWorlds()) {
```

Comment:
If a player sleeps in a bed in a modded dim, will this update the overworld? As im not 100% sure it will.

Reply:
It works as expected in a LAN devenv world where the other player is in the overworld and the other player is in a custom dimension.

Reply:
... but there's another bug -- only the overworld players seem to get checked when the hosting LAN player is in the overworld.

Reply:
Should I hack around `ServerWorld.updatePlayersSleeping` or just remove the mixin?

Reply:
I don't remember what that hack around `ServerWorld.updatePlayersSleeping` would be, so I'll make this PR only have the `isDay`/`isNight` stuff.